machine_learning_hep/analysis/analyzerdhadrons_mult.py

```diff
@@ -433,23 +452,27 @@ def makenormyields(self):
         self.loadstyle()
         #self.test_aliphysics()
         #filedataval = TFile.Open(self.f_evtnorm)
-        fileouteff = "%s/efficiencies%s%s.root" % \
-                      (self.d_resultsallpmc, self.case, self.typean)
         yield_filename = self.make_file_path(self.d_resultsallpdata, self.yields_filename, "root",
                                              None, [self.case, self.typean])
         gROOT.LoadMacro("HFPtSpectrum.C")
         from ROOT import HFPtSpectrum, HFPtSpectrum2, HFPtSpectrumRescaled
         histonorm = TH1F("histonorm", "histonorm", self.p_nbin2, 0, self.p_nbin2)
         for imult in range(self.p_nbin2):
+            fileouteff = "%s/efficiencies%s%s.root" % \
+                          (self.d_resultsallpmc, self.case, self.typean) \
+                          if not self.path_file_eff[imult]  else self.path_file_eff[imult]
+            if not os.path.exists(fileouteff):
+                self.logger.fatal("Efficiency file %s could not be found", fileouteff)
             bineff = -1
             if self.p_bineff is None:
                 bineff = imult
                 print("Using efficiency for each var2 bin")
             else:
                 bineff = self.p_bineff
                 print("Using efficiency always from bin=", bineff)
```

Comment:
This print statement will be a bit confusing, as your changes come afterwards

Reply:
I don't really understand what you mean here. This print statement should be here because it is the first place where it can be checked whether it is present or not. And it cannot come before as the path might depend in `imult`.

Reply:
I meant the ```print("Using efficiency for each var2 bin")``` and ```print("Using efficiency always from bin=", bineff)``` statements, in combination with the ```mult_bin_eff = bineff if self.mult_bin_eff[imult] is None else self.mult_bin_eff[imult]``` line below. So first you print based on the ```self.p_bineff``` variable, but this one is not used when you have set the ```self.mult_bin_eff[imult]``` var. So can be confusing. (minor comment :) )

Reply:
Nono, you are definitely right. I'll make that coherent